### PR TITLE
Add grid functionality

### PR DESCRIPTION
**Pull request by jaredpereira**
Hi! This is a small PR to add grid_functionality. Most of the math was copied over from `earthsea`, and I did my best to reuse the functions already present in `molly_the_polly` and to follow the style.

Let me know if this is handy! Happy to iterate to get it to fit the code base :)

### CODE REVIEW

File: molly_the_poly.lua [hunk 1]
```diff
@@ -18,6 +18,7 @@ local SCREEN_FRAMERATE = 15
 local screen_dirty = true
 
 local midi_in_device
+local g
```
Comment (markwheeler) [inline molly_the_poly.lua]:
```suggestion
local grid_device
```

File: molly_the_poly.lua [hunk 2]
```diff
@@ -149,6 +150,22 @@ function key(n, z)
...
+
+  if z == 1 then
+    note_on(note_num, 100)
+    g:led(x, y,15)
+  else
+    g:led(x, y,0)
+    note_off(note_num)
+  end
+  g:refresh()
+end
```
Comment (markwheeler) [inline molly_the_poly.lua]: Suggesting some renaming, no need to translate to freq and fixed note velocity (0-1 range):

```suggestion
local function grid_key(x, y, z)
  local note_num = util.clamp(((7 - y) * 5) + x + 33, 0, 127)
  
  if z == 1 then
    note_on(note_num, 0.8)
    grid_device:led(x, y, 15)
  else
    note_off(note_num)
    grid_device:led(x, y, 0)
  end
  grid_device:refresh()
end
```

File: molly_the_poly.lua [hunk 3]
```diff
@@ -239,6 +256,10 @@ function init()
   midi_in_device = midi.connect(1)
   midi_in_device.event = midi_event
   
+
+  g = grid.connect()
```
Comment (markwheeler) [inline molly_the_poly.lua]:
```suggestion
  grid_device = grid.connect(1)
```

File: molly_the_poly.lua [hunk 4]
```diff
@@ -239,6 +256,10 @@ function init()
...
+  g.key = grid_event
```
Comment (markwheeler) [inline molly_the_poly.lua]:
```suggestion
  grid_device.key = grid_key
```

**Review by markwheeler — CHANGES_REQUESTED** [review]
Nice addition, thanks! I'm suggesting some changes for consistency and a little simplification regarding calculating the note number. Sorry it took me so long to look this over!

**Comment by jaredpereira** [comment]
Whoops! Sorry thought that your fixes were merged in automatically. Everything made sense to me, so merged em all in. Much shorter on calculating the note number,

**Comment by markwheeler** [comment]
thank you!

**Pull request merged**